File: Examples/Features/RangeOperations/GetRangeBoundary.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Text;

namespace GrapeCity.Documents.Excel.Examples.Features.RangeOperations
{
    public class GetRangeBoundary : ExampleBase
    {
        public override void Execute(GrapeCity.Documents.Excel.Workbook workbook)
        {
            var fileStream = this.GetResourceStream("xlsx\\Sport sign-up sheet.xlsx");
            workbook.Open(fileStream);

            IWorksheet worksheet = workbook.Worksheets[0];


            // Get the absolute location and size of the Range["G1"] in the worksheet.
            IRange range = worksheet.Range["G1"];
            Rectangle rect = GrapeCity.Documents.Excel.CellInfo.GetRangeBoundary(range);
            // Add the image to the Range["G1"]
            System.IO.Stream stream = this.GetResourceStream("logo.png");
            worksheet.Shapes.AddPictureInPixel(stream, GrapeCity.Documents.Excel.Drawing.ImageType.PNG, rect.X, rect.Y, rect.Width, rect.Height);
        }

        public override string[] UsedResources
        {
            get
            {
                return new string[] { "xlsx\\Sport sign-up sheet.xlsx", "logo.png" };
            }
        }

        public override bool IsNew
        {
            get
            {
                return true;
            }
        }
    }
}

File: Examples.VB/Features/RangeOperations/GetRangeBoundary.vb
Namespace Features.RangeOperations
    Public Class GetRangeBoundary
        Inherits ExampleBase
        Public Overrides Sub Execute(workbook As Excel.Workbook)
            'Open an excel file
            Dim fileStream = GetResourceStream("xlsx\\Sport sign-up sheet.xlsx")
            workbook.Open(fileStream)

            Dim worksheet As IWorksheet = workbook.Worksheets(0)

            'Get the absolute location And size of the Range["G1"] in the worksheet.
            Dim range As IRange = worksheet.Range("G1")
            Dim rect As System.Drawing.Rectangle = Excel.CellInfo.GetRangeBoundary(range)
            'Add the image to the Range["G1"]
            Dim stream As IO.Stream = GetResourceStream("logo.png")
            worksheet.Shapes.AddPictureInPixel(stream, ImageType.PNG, rect.X, rect.Y, rect.Width, rect.Height)
        End Sub

        Public Overrides ReadOnly Property UsedResources() As String()
            Get
                Return New String() {"xlsx\\Sport sign-up sheet.xlsx", "logo.png"}
            End Get
        End Property
    End Class
End Namespace
